README.md
Changes in the last version Astrolog 5.41G

   * Fixed original Astrolog bug, where Local Horizon data (and also Prime
     Vertical) were off by up to several arc-minutes.

   * Fixed original Astrolog bug (Windows version), where "Time / Space
     Midpoint" chart is correct, but it was not possible to return back to
     "No relationship chart" - there was still previous "midpoint" chart.

   * In "Rising and Setting" chart fixed bug, where "Print Nearest Second"
     worked unproperly - seconds weren't shown for risings and settings.

     In this context more correct termin instead of "zenith" and "nadir" is
     astronomical termin "culmination" that means transit over meridian.
     There are two culminations: "upper culmination" when planet has
     highest position and "lower culmination" when planet has lowest
     position. So in the chart termins "zeniths" and "nadirs" are replaced
     by "culm.(up)" and "culm.(lo)" respectively.

   * Uncorrect object name "Nadir" has been replaced by corret "IC". Nadir
     has different meaning than IC, Nadir is the point on the sky opposite
     to Zenith.

   * Slightly improved Astrolog's behaviour in case of progressed and
     relationship charts. Earlyer Windows versions allowed to do
     progression and then do some relationship chart like Natal-Progressed.
     Or vice versa, when looking say Natal-Progressed chart switch also
     Progression "on". That all confused Astrolog and results were often
     unpredictable.

     Now any use of relationship charts (in menu "Info") switches off "Do
     Progression" (in "Chart" => "Progressions"). And vice versa, switching
     progression on (in "Chart" => "Progression") switches off any
     comparison. As result Astrolog is displaying just what it has to
     dasplay, no confusion.

   * Fixed orignal astrolog bug (appeared only in Windows version). When
     natal chart was opened from file with saved planets positions (not
     birth data), there only first output was correct, all next charts (as
     aspect list, transit searches, all comparison charts etc) had wrong
     planets positions.

     NB! Use such input files carefully! They are treated by astrolog as
     files "with no time and space", so a lot of chart types can't be used
     with them. An examples are all progression charts, because they needs
     birth-chart time to calculate progressions.

   * Fixed bug of changed versions, where in case of use of data with saved
     planet positions (just as above), data-border of graphics charts
     dilsplayed info "no time or space" twice.

   * Time and location have new format - with seconds, both for chart's
     data inputs and outputs. Only outputs with old data format are events
     searching outputs, because with reasonably low division value they
     anyway can be off by a few minutes, so seconds seems unnecessary
     there.

          "Rising and setting" chart is only exception among searching
          charts that shows time with seconds. Even with default setting d:
          48 times are off by only a few seconds, and it is quick enough to
          show precise (by seconds) times with d: 96 in reasonably short
          time. Note, that values of azimuth angles in rising/setting
          moment are extremely sensitive of time changes (very small change
          of time causes big change in azimuth) and can be off by a few
          seconds. To improve azimuth precision one has to increase
          division value up to hundreds.

     All input files with old time/location format can be used, but all
     outputs are saved only in new time/location format.

   * Calculations precision improved - time-dependent obliquity of ecliptic
     is used instead of fixed value.

   * Added new switsh to change local horizon text output from default ENWS
     to NESW. One has just to add line

          =YZ

     to astrolog.dat file and it will be deafult setting.

   * Slightly changed Prime Vertical text chart output: Altitude and
     Azimuth in header have been replaced by Amplitude and Prm-Vrt. One can
     see this chart by choosing "Local Horizon" and setting "Horizon Chart
     with Polar Center / Prime Vert. (text)" in "Chart settings". Note,
     that this switch has different meanings for text and graphics charts:
     in text mode it switches between Local Horizon <=> Prime Vertical
     outputs, in graphics it switches between "normal wiew" and "view vith
     Polar Center". However, one can switch also between Polar Center and
     Pime Vert. also in graphics by hitting 'i' on the keyboard (switch to
     bonus mode).

   * Changed calculations of Solar Arc. Original Astrolog calculates there
     actually directions, where all planets and house cusps positions are
     moved forward to an amount equal in degrees to the number of years
     that have passed between the specified date and the chart in question.
     Because real (even mean or average) motion of Sun isn't 1 degree per
     day, resulting Solar Arc Sun position doesn't match with secondary
     progressed Sun position (but they must match).

     To correct this situation, there has been added another calculation of
     Solar Arc - first secondary progressed Sun's position is determined
     and then all chart components are moved accordlingly.

     To avoid misunderstanding, existing Solar Acr calculations remains,
     but have now corrected name "Degree Per Day/Month", as earlyer one can
     change amount of degrees per year (more strictly, amount of days for 1
     degree direction. See description of -p0 and -pd switches in
     helpfile.540). Added calculation is named "Solar Arc Directions" and
     there Sun's position follows secondary progressed Sun positions, all
     other point in chart are moved accordlingly.

     Existing switches -p0, -p0n remain as previously. For new correct
     "Solar Arc Directions" calculations new switches -p1 and -p1n are
     used. In Windows version all can be done through menus: "Chart" =>
     "Progressions..."

   * Slightly changed chraphics charts' infoborder and text charts' headers
     - secondary progressed charts are still named as "progressed", Solar
     Arc and "Degree per Year/Mont" are marked as "directed" with name of
     direction.

   * Graphics chart's infoborder has additional information: Obliquity of
     ecliptic, Sidereal time, Delta T (in seconds) and in case of sidereal
     chart Ayanamsha (sidereal offest).

     Note, that Ayanamsha has negative value and dafaults to Fagan Bradley.
     Ayanamsha control in "Calculation Settings" has a dropdown to allow
     quick selection of some common systems of sidereal astrology. The
     values are additions to default value and they are 0.0 for Fagan
     Bradley, 0.883333 (or 0 degrees 53') for N.C. Lahiri, 0.983333 (or 0
     degrees 59') for Krishnamurti, and 2.333333 (or 2 degrees 20') for
     B.V. Raman. On the screen has shown resulting value.

   * Date/time and Julian Day in the graphics charts' infoboredr has been
     colored:

          1. First (or single) data/time is always bright white.

          2. Second date/time is always yellow.

          3. JD (which is actually also time/date) has color of
          corresponding date/time above (as yellow in case of transit
          comparison or direction charts). If JD doesn't correspond to
          neither date/time above (as in case of progressions where JD
          corresponds to date/time of planets' positions on the screen),
          color is green. Note, that in case of Synastry and Comparison
          charts JD will be yellow, becsuse it corresponds to second
          date/time above (data of second chart).

   * Windows version controls of progressed charts have been polished. They
     looks more clear now. As default 365.2422 (tropical) year has used.

     In "Degree per Days" dropdown are also available:

          365.25636 (sidereal year)
          27.321582 (tropical month)
          27.321661 (sidereal month)
          29.530588 (synodic month)

     Other more exotic (like Draconic) years/months aren't included.

   * Astrolog computes position of Lilith (Dark Moon) using external
     ephemeris. When ephemeris are set off, Astrolog will display the
     position of the South Node instead (see helpfile.540, description of
     -HO switch). For users, who wants to use South Node always, there has
     been added new switch =YN which forces Astrolog to do it. This switch
     can be entered through "Edit" => "Enter Command Line" (Windows
     version) or simply added to astrolog.dat file to make this behaviour
     default.

   * By default Astrolog calculates mean Lilith. For users who wants to use
     osculting position of Lilith, new switch -YL has been added. As above,
     this switch can be added to astrolog.dat file, which makes such
     behaviour default.

   * Dispositors glyphs on the graphics wheels can be switched on/off
     using switch -YD. It can be added to astrolg.dat file: =YD (on) or
     _YD (off), default "on".

---------------------------------------------------------------------------

18th. May. 2002





changes.txt
Astrolog 5.41G - overview of changes.

Detailed description of changes I have made in original version of
Astrolog, are given in history of changes. But there changes are in
chronological order and it's difficult to see main changes. Here changes
are devided between different areas, which helps to see, what's made.

Alois Treindl modified Astrolog 5.40 and named new version 5.41. It
contains the new Swiss Ephemeris instead of the older Placalc ephemeris.
The responsibility for this modification is with Alois Treindl. Original
Astrolog 5.41 with all related files is available at
ftp://www.astro.com/pub/astrolog/.

I used Astrolog 5.41 as a base, where are moved my changes, made in
previous versions.


                             Table of Contents

                      Back to Valja's astrology page.
                      Fixed Astrolog 5.30 - 5.40 bugs
                            Houses in Polar Zone
                             Added new objects
                             Power calculations
                     Handling of Stars and Arabic Parts
                       Animation of progressed charts
                            Changes in graphics
                             "Flowers" graphics
                            Calculation options
                            Customized rulership
                          8-bit characters support
                         Alternative interpretation
                           Miscellaneous changes

  ------------------------------------------------------------------------
*Fixed Astrolog 5.30 - 5.40 bugs

   * Corrected calculation of Alcabitius house cusps. Thanks to Per Dahlin
     (capella@algonet.se) who sent me definition of Alcabitius house cusps.

   * Fixed original Astrolog omission, where in case of -sr switch
     (equatorial coordinates instead ecliptical) coordinates of house cusps
     aren't transformed from ecliptics to equator, and planets itself was
     on equator, but was placed to ecliptical houses.

   * All -Z switch (Horizon) charts must be independent from -sr switch,
     because image of visible sky relatively to local horizon must be
     independent of coordinate system we use for planets (equatorial,
     ecliptical or some other). In original Astrolog all outputs of -Z
     switches changes in case of additional switch -sr. All routines,
     connected with -Z switch, transforms first coordinates from ecliptic
     to equator and then from equator to local horizon coordinates. In case
     of -sr switch first transform from ecliptic to equator is superfluous,
     because in case of -sr switch coordinates are already given in
     equatorial system. Since 5.30b this bug is fixed.

   * In Windows version using of parallel-contraparallel aspects can be set
     very simply. But Transit To Transit and Transit To Natal Hits are
     calculated for normal aspects independent of setting of Parallel
     aspect switch. Only effect is change of aspect names from conjunctions
     and Oppositions to Parallels and Contraparallels, but used are really
     normal aspects, also trines, squares etc. In original Astrolog
     searching of parallel and contraparallel hits is absent at all. In
     changed version this omissed part is added and is possible to search
     also Parallel and ContraParallel aspects like it was possible with
     normal aspects.

   * Fixed bug, where applayng-separating of comparison charts natal -
     progressed and natal - transit was wrong. Now with -rt and -rp switch
     natal chart planets velocities are made zero, and -rt (or -rp) with -a
     switch give correct applay-separate aspects (like -T switch).

   * Fixed bug in calculation of Gauquelin sectors in polar zone. In this
     case some planets can not rise and set during a day. Astrolog
     restricts such planets, but original version don't restores old
     restriction set when returning to standard wheel.

   * Fixed original Astrolog bug, appeared between versions 5.30 and 5.40 -
     positions of planets and Houses in Gauquelin graphics chart infoborder
     was wrong.

   * Fixed original Astrolog omission - Windows version "Help" menu looks
     for necessary files only in local directory. Now these files are
     searched through all directories, connected with astrolog (see
     helpfile for details of setting of environment variables) with
     searching order:

        o 1) Local directory
        o 2) Environment directories, set in autoexec.bat, one of them
          version specific (names of directories as examples, names of
          environment variables must be strict and in upper-case:
             + a) set ASTR5.41B=C:\aaaaa\541
             + b) set ASTROLOG=d:\abbbb
             + c) set ASTR=H:\miscfile
        o 3) Default directory, compiled in: C:\ASTROLOG

     If Astrolog doesn't find file, it doesn't attempt to start wordpad or
     WEB browser but prints only warning, that it can't find file.

   * Fixed original astrolog bug. POF (in main objects) was wrong in case
     of equatorial coordinates (but in arabic parts list it was correct).

   * Fixed original Astrolog bug, where Local Horizon data (and also Prime
     Vertical) were off by up to several arc-minutes.

   * Fixed original Astrolog bug (Windows version), where "Time / Space
     Midpoint" chart is correct, but it was not possible to return back to
     "No relationship chart" - there was still previous "midpoint" chart.

   * In "Rising and Setting" chart fixed bug, where "Print Nearest Second"
     worked unproperly - seconds weren't shown for risings and settings.

     In this context more correct termin instead of "zenith" and "nadir" is
     astronomical termin "culmination" that means transit over meridian.
     There are two culminations: "upper culmination" when planet has
     highest position and "lower culmination" when planet has lowest
     position. So in the chart termins "zeniths" and "nadirs" are replaced

data2.c
/*
** Astrolog (Version 5.41G) File: data2.c
**
** Code changed by Valentin Abramov
**
** IMPORTANT NOTICE: The graphics database and chart display routines
** used in this program are Copyright (C) 1991-1998 by Walter D. Pullen
** (Astara@msn.com, http://www.magitech.com/~cruiser1/astrolog.htm).
** Permission is granted to freely use and distribute these routines
** provided one doesn't sell, restrict, or profit from them in any way.
** Modification is allowed provided these notices remain with any
** altered or edited versions of the program.
**
** The main planetary calculation routines used in this program have
** been Copyrighted and the core of this program is basically a
** conversion to C of the routines created by James Neely as listed in
** Michael Erlewine's 'Manual of Computer Programming for Astrologers',
** available from Matrix Software. The copyright gives us permission to
** use the routines for personal use but not to sell them or profit from
** them in any way.
**
** The PostScript code within the core graphics routines are programmed
** and Copyright (C) 1992-1993 by Brian D. Willoughby
** (brianw@sounds.wa.com). Conditions are identical to those above.
**
** The extended accurate ephemeris databases and formulas are from the
** calculation routines in the library SWISS EPHEMERIS and are programmed and
** copyright 1998 by Astrodienst AG.
** The use of that source code is subject to
** the Swiss Ephemeris Public License, available at 
** http://www.astro.ch/swisseph. This copyright notice must not be 
** changed or removed by any user of this program.
**
** Initial programming 8/28,30, 9/10,13,16,20,23, 10/3,6,7, 11/7,10,21/1991.
** X Window graphics initially programmed 10/23-29/1991.
** PostScript graphics initially programmed 11/29-30/1992.
** Last code change made 12/20/1998.
** Modifications from version 5.40 to 5.41 are by Alois Treindl.
*/

#include "astrolog.h"


/*
******************************************************************************
** Constellation Tables.
******************************************************************************
*/

#ifdef CONSTEL
CONST byte * ARR szCnstlName[cCnstl+1] = {"",
  "Andromeda", "Antilia", "Apus", "Aquarius",
  "Aquila", "Ara", "Aries", "Auriga",
  "Bootes", "Caelum", "Camelopardalis", "Cancer",
  "Canes Venatici", "Canis Major", "Canis Minor", "Capricornus",
  "Carina", "Cassiopeia", "Centaurus", "Cepheus",
  "Cetus", "Chamaeleon", "Circinus", "Columba",
  "Coma Berenices", "Corona Australis", "Corona Borealis", "Corvus",
  "Crater", "Crux", "Cygnus", "Delphinus",
  "Dorado", "Draco", "Equuleus", "Eridanus",
  "Fornax", "Gemini", "Grus", "Hercules",
  "Horologium", "Hydra", "Hydrus", "Indus",
  "Lacerta", "Leo", "Leo Minor", "Lepus",
  "Libra", "Lupus", "Lynx", "Lyra",
  "Mensa", "Microscopium", "Monoceros", "Musca",
  "Norma", "Octans", "Ophiuchus", "Orion",
  "Pavo", "Pegasus", "Perseus", "Phoenix",
  "Pictor", "Pisces", "Piscis Austrinus", "Puppis",
  "Pyxis", "Reticulum", "Sagitta", "Sagittarius",
  "Scorpius", "Sculptor", "Scutum", "Serpens Caput/Cauda",
  "Sextans", "Taurus", "Telescopium", "Triangulum",
  "Triangulum Australe", "Tucana", "Ursa Major", "Ursa Minor",
  "Vela", "Virgo", "Volans", "Vulpecula"};

CONST byte * ARR szCnstlAbbrev[cCnstl+1] = {"",
  "And", "Ant", "Aps", "Aqu", "Aql", "Ara", "Ari", "Aur",
  "Boo", "Cae", "Cam", "Cnc", "CVn", "CMa", "CMi", "Cap",
  "Car", "Cas", "Cen", "Cep", "Cet", "Cha", "Cir", "Col",
  "Com", "CrA", "CrB", "Crv", "Crt", "Cru", "Cyg", "Del",
  "Dor", "Dra", "Equ", "Eri", "For", "Gem", "Gru", "Her",
  "Hor", "Hya", "Hyi", "Ind", "Lac", "Leo", "LMi", "Lep",
  "Lib", "Lup", "Lyn", "Lyr", "Men", "Mic", "Mon", "Mus",
  "Nor", "Oct", "Oph", "Ori", "Pav", "Peg", "Per", "Phe",
  "Pic", "Psc", "PsA", "Pup", "Pyx", "Ret", "Sge", "Sgr",
  "Sco", "Scl", "Sct", "Ser", "Sex", "Tau", "Tel", "Tri",
  "TrA", "Tuc", "UMa", "UMi", "Vel", "Vir", "Vol", "Vul"};

CONST byte * ARR szCnstlMeaning[cCnstl+1] = {"",
  "Chained Maiden", "Air Pump", "Bird of Paradise", "Water Bearer",
  "Eagle", "Altar", "Ram", "Charioteer",
  "Herdsman", "Chisel", "Giraffe", "Crab",
  "Hunting Dogs", "Great Dog", "Little Dog", "Sea Goat",
  "Keel", "Queen", "Centaur", "King",
  "Sea Monster", "Chameleon", "Drawing Compass", "Dove",
  " Berenice's Hair", "Southern Crown", "Northern Crown", "Crow",
  "Cup", "Southern Cross", "Swan", "Dolphin",
  "Dolphinfish", "Dragon", "Little Horse", "River",
  "Furnace", "Twins", "Crane", "Strongman",
  "Clock", "Sea Serpent", "Water Snake", "Indian",
  "Lizard", "Lion", "Little Lion", "Hare",
  "Scales", "Wolf", "Lynx", "Lyre",
  "Table Mountain", "Microscope", "Unicorn", "Fly",
  "Level", "Octant", "Serpent Bearer", "Hunter",
  "Peacock", "Winged Horse", "Hero", "Phoenix",
  "Painter", "Fishes", "Southern Fish", "Stern",
  "Compass", "Net", "Arrow", "Archer",
  "Scorpion", "Sculptor", "Shield", " Head/Tail of the Snake",
  "Sextant", "Bull", "Telescope", "Triangle",
  "Southern Triangle", "Toucan", "Great Bear", "Little Bear",
  "Sail", "Virgin", "Flying Fish", "Fox"};

CONST byte * ARR szCnstlGenitive[cCnstl+1] = {"",
  "", "", "2odis", "", "", "", "1tis", "",
  "2is", "", "", "2ri", "2um1orum", " is", " is", "",
  "", "", "", "", "", "tis", "", "",
  "", "", "", "", "is", "", "", "",
  "1us", "nis", "", "", "", "1orum", "1is", "2is",
  "", "", "", "", "", "nis", "nis0is", "2oris",
  "", "", "", "", "", "", "1tis", "",
  "", "1is", "", "is", "nis", "", "", "",
  "is", "2ium", "", "", "1dis", "", "", "",
  "", "is", "", "!1is", "1tis", "", "", "",
  "2i1is", "", " is", " is", "1orum", "1inis", "1tis", ""};
#endif /* CONSTEL */


/*
******************************************************************************
** Object Calculation Tables.
******************************************************************************
*/

#ifdef MATRIX
CONST byte rErrorCount[oPlu-oJup+1] = {11, 5, 4, 4, 4};
CONST byte rErrorOffset[oPlu-oJup+1] = {0, 72, 72+51, 72+51+42, 72+51+42+42};

CONST real ARR rErrorData[72+51+42*3] = {
-.001,-.0005,.0045,.0051,581.7,-9.7,-.0005,2510.7,-12.5,-.0026,1313.7,-61.4,
0.0013,2370.79,-24.6,-.0013,3599.3,37.7,-.001,2574.7,31.4,-.00096,6708.2,
-114.5,-.0006,5499.4,-74.97,-.0013,1419,54.2,.0006,6339.3,-109,.0007,4824.5,
-50.9,.0020,-.0134,.0127,-.0023,676.2,.9,.00045,2361.4,174.9,.0015,1427.5,
-188.8,.0006,2110.1,153.6,.0014,3606.8,-57.7,-.0017,2540.2,121.7,-.00099,
6704.8,-22.3,-.0006,5480.2,24.5,.00096,1651.3,-118.3,.0006,6310.8,-4.8,.0007,
4826.6,36.2, /* Jupiter error */

-.0009,.0037,0,.0134,1238.9,-16.4,-.00426,3040.9,-25.2,.0064,1835.3,36.1,
-.0153,610.8,-44.2,-.0015,2480.5,-69.4,-.0014,.0026,0,.0111,1242.2,78.3,
-.0045,3034.96,62.8,-.0066,1829.2,-51.5,-.0078,640.6,24.2,-.0016,2363.4,
-141.4,.0006,-.0002,0,-.0005,1251.1,43.7,.0005,622.8,13.7,.0003,1824.7,-71.1,
.0001,2997.1,78.2, /* Saturn error */

-.0021,-.0159,0,.0299,422.3,-17.7,-.0049,3035.1,-31.3,-.0038,945.3,60.1,
-.0023,1227,-4.99,.0134,-.02186,0,.0317,404.3,81.9,-.00495,3037.9,57.3,.004,
993.5,-54.4,-.0018,1249.4,79.2,-.0003,.0005,0,.0005,352.5,-54.99,.0001,3027.5,
54.2,-.0001,1150.3,-88, /* Uranus error */

0.1832,-.6718,.2726,-.1923,175.7,31.8,.0122,542.1,189.6,.0027,1219.4,178.1,
-.00496,3035.6,-31.3,-.1122,.166,-.0544,-.00496,3035.3,58.7,.0961,177.1,-68.8,
-.0073,630.9,51,-.0025,1236.6,78,.00196,-.0119,.0111,.0001,3049.3,44.2,-.0002,
893.9,48.5,.00007,1416.5,-25.2, /* Neptune error */

-.0426,.073,-.029,.0371,372,-331.3,-.0049,3049.6,-39.2,-.0108,566.2,318.3,
0.0003,1746.5,-238.3,-.0603,.5002,-.6126,.049,273.97,89.97,-.0049,3030.6,61.3,
0.0027,1075.3,-28.1,-.0007,1402.3,20.3,.0145,-.0928,.1195,.0117,302.6,-77.3,
0.00198,528.1,48.6,-.0002,1000.4,-46.1 /* Pluto error */
};

OE ARR rgoe[oVes-1+cUran] = {
{358.4758,35999.0498,-.0002,.01675,-.4E-4,0,1,101.2208,1.7192,.00045,0,0,0,0,
0,0}, /* Earth/Sun */
{102.2794,149472.515,0,.205614,.2E-4,0,.3871,28.7538,.3703,.0001,47.1459,
1.1852,0.0002,7.009,.00186,0}, /* Mercury */
{212.6032,58517.8039,.0013,.00682,-.5E-4,0,.7233,54.3842,.5082,-.14E-2,
75.7796,0.8999,.4E-3,3.3936,.1E-2,0}, /* Venus */
{319.5294,19139.8585,.2E-3,.09331,.9E-4,0,1.5237,285.4318,1.0698,.1E-3,
48.7864,0.77099,0,1.8503,-.7E-3,0}, /* Mars */

{225.4928,3033.6879,0,.04838,-.2E-4,0,5.2029,273.393,1.3383,0,99.4198,1.0583,
0,1.3097,-.52E-2,0}, /* Jupiter */
{174.2153,1223.50796,0,.05423,-.2E-3,0,9.5525,338.9117,-.3167,0,112.8261,
.8259,0,2.4908,-.0047,0}, /* Saturn */
{74.1757,427.2742,0,.04682,.00042,0,19.2215,95.6863,2.0508,0,73.5222,.5242,0,
0.7726,.1E-3,0}, /* Uranus */
{30.13294,240.45516,0,.00913,-.00127,0,30.11375,284.1683,-21.6329,0,130.68415,
1.1005,0,1.7794,-.0098,0}, /* Neptune */
{229.781,145.1781,0,.24797,.002898,0,39.539,113.5366,.2086,0,108.944,1.3739,0,
17.1514,-.0161,0}, /* Pluto */

{34.6127752,713.5756219,0,.382270369,-.004694073,0,13.66975144,337.407213,
2.163306646,0,208.1482658,1.247724355,0,6.911179715,.011236955,0}, /* Chiron */
{108.2925,7820.36556,0,.0794314,0,0,2.7672273,71.0794444,0,0,80.23555556,
1.3960111,0,10.59694444,0,0}, /* Ceres */
{106.6641667,7806.531667,0,.2347096,0,0,2.7704955,310.166111,0,0,172.497222,
1.39601111,0,34.81416667,0,0}, /* Pallas Athena */
{267.685,8256.081111,0,.2562318,0,0,2.6689897,245.3752778,0,0,170.137777,
1.396011111,.0003083333,13.01694444,0,0}, /* Juno */
{138.7733333,9924.931111,0,.0902807,0,0,2.360723,149.6386111,0,0,103.2197222,
1.396011111,.000308333,7.139444444,0,0}, /* Vesta */

{104.5959,138.5369,0,0,0,0,40.99837, 0,0,0,0,0,0,0,0,0}, /* Cupido   */
{337.4517,101.2176,0,0,0,0,50.667443,0,0,0,0,0,0,0,0,0}, /* Hades    */
{104.0904,80.4057, 0,0,0,0,59.214362,0,0,0,0,0,0,0,0,0}, /* Zeus     */
{17.7346, 70.3863, 0,0,0,0,64.816896,0,0,0,0,0,0,0,0,0}, /* Kronos   */
{138.0354,62.5,    0,0,0,0,70.361652,0,0,0,0,0,0,0,0,0}, /* Apollon  */
{-8.678,  58.3468, 0,0,0,0,73.736476,0,0,0,0,0,0,0,0,0}, /* Admetos  */
{55.9826, 54.2986, 0,0,0,0,77.445895,0,0,0,0,0,0,0,0,0}, /* Vulkanus */
{165.3595,48.6486, 0,0,0,0,83.493733,0,0,0,0,0,0,0,0,0}, /* Poseidon */
{170.73,  51.05,   0,0,0,0,79.22663, 0,0,0,0,0,0,0,0,0}  /* Proserpina */
};
#endif /* MATRIX */

real ARR rStarBright[cStar+1] = {0,
  0.46, 2.02, 5.24, 5.09, 0.85, 0.08, 0.12, 1.64, 1.65, 1.70,
  0.50, 1.90, 1.98,-0.72, 1.93,-1.46, 1.50, 1.86, 1.58, 0.38,
  1.14, 1.82, 1.86, 1.68, 1.98, 1.35, 1.79, 1.58, 1.63, 1.25,
  1.77, 0.97, 1.86, 0.61,-0.04,-0.01, 0.96, 1.63, 1.87, 1.85,
  0.03, 0.77, 1.94, 1.25, 1.74, 1.16, 4.61,
  2.06, 2.83, 2.23, 2.29, 4.33, 2.25, 2.12, 2.23, 3.53, 4.25,
  2.37, 2.83, 2.27, 2.08, 2.75, 2.61, 2.23, 2.65, 2.08, 3.24,
  2.44, 3.68, 3.27, 2.42, 2.49, 9.99, 9.99};

CONST real ARR rStarData[cStar*6] = {
 1,37,42.9,-57,14,12,   2,31,50.5, 89,15,51,   3,17,46.1,-62,34,32,
 3,49,11.1, 24, 8,12,   4,35,55.2, 16,30,33,   5,16,41.4, 45,59,53,
 5,14,32.3, -8,12, 6,   5,25, 7.9,  6,20,59,   5,26,17.5, 28,36,27,
 5,36,12.8, -1,12, 7,   5,55,10.3,  7,24,25,   5,59,31.7, 44,56,51,
 6,22,42.0,-17,57,21,   6,23,57.1,-52,41,45,   6,37,42.7, 16,23,57,
 6,45, 8.9,-16,42,58,   6,58,37.6,-28,58,20,   7, 8,23.5,-26,23,36,
 7,34,36.0, 31,53,19,   7,39,18.1,  5,13,30,   7,45,18.9, 28, 1,34,
 8, 9,32.0,-47,20,12,   8,22,30.8,-59,30,35,   9,13,12.0,-69,43, 2,
 9,27,35.2, -8,39,31,  10, 8,22.3, 11,58, 2,  11, 3,43.7, 61,45, 3,
12,26,35.9,-63, 5,57,  12,31, 9.9,-57, 6,48,  12,47,43.2,-59,41,19,
12,54, 1.7, 55,57,35,  13,25,11.6,-11, 9,41,  13,47,32.4, 49,18,48,
14, 3,49.4,-60,22,23,  14,15,39.7, 19,10,57,  14,39,35.9,-60,50, 7,
16,29,24.4,-26,25,55,  17,33,36.5,-37, 6,14,  17,37,19.2,-42,59,52,
18,24,10.3,-34,23, 5,  18,36,56.3, 38,47, 1,  19,50,47.0,  8,52, 6,
20,25,38.9,-56,44, 6,  20,41,25.9, 45,16,49,  22, 8,14.0,-46,57,40,
22,57,39.0,-29,37,20,   0,42, 7.0, 41,16, 0,   0, 8,23.3, 29,05,26,
 0,13,14.0, 15,11, 1,   0,40,30.4, 56,32,15,   1, 9,43.9, 35,37,14,
 2, 2, 2.7,  2,45,51,   2, 3,54.0, 42,19,47,   3, 8,10.1, 40,57,20,
 5,32, 0.4,-0,-17,57,   7,20, 7.4, 21,58,56,   8,58,29.2, 11,51,28,
11, 1,50.5, 56,22,57,  13, 2,10.6, 10,57,33,  13,23,55.5, 54,55,32,
14,50,42.3, 74, 9,20,  14,50,52.7,-16, 2,30,  15,17, 0.4, -9,22,59,
15,34,41.3, 26,42,53,  15,44,16.0,  6,25,32,  17,34,56.0, 12,33,36,
19,30,43.3, 27,57,35,  21,18,34.8, 62,35, 8,  21,40, 5.5,-16,39,44,
22,54,39.0,-15,49,15,  23, 3,46.5, 28, 4,58,  23, 4,45.7, 15,12,19,
17,45,37.42,-29, 0,21,  18, 4, 0.0, 30, 0, 0};

#ifdef INTERPRET
/*
******************************************************************************
** Interpretation Tables.
******************************************************************************
*/

byte * ARR szMindPart[oNorm+1] = {"earthy nature",
  "spirit, ego, image of self, and sense of aliveness",
  "emotional nature, moods, feelings, and caring tendencies",
  "thoughts, intellect, and communicative activity",
  "creativity, tendencies for affection and calmness, and relationship needs",
  "will, energy, activity, and aggressive, assertive tendencies",
  "enthusiastic, faithful, wise, expansive, spontaneous nature",
  "part of mind that is disciplined, respectful, and solitary",
  "individuality, desires for change, and tendency to go against social norms",
  "intuitive, spiritual, compassionate, psychic nature",
  "destiny, and capacity to transform the self and the outer world",
  "ability to help, heal, and teach others, and where one has much experience",
  "tendency to direct energy to instinctive, creative, reproductive goals",
  "tendency to direct energy away from emotional and into mental pursuits",
  "tendency to give away individual power for the benefit of a relationship",
  "capacity to direct creative energy into personal or devotional goals",
  "karmic goals, and best direction of evolutionary growth",
  "capacity to let go and accept their natural darker side",
  "place where opportunity and success can be found",
  "karmic fate and method of reception of energies",
  "generalized projected personality and outward appearance",
  "personality and outward appearance, as projected to the world",
  "", "", "", "", "", "", "", "",
  "view as seen in the eyes of others, reputation, and social standing",
  "", "",
  "group associations and matters relating to art",
  "dark, secretive, past, shameful side",
  "fiery, creative, leadership tendencies",
  "lofty, extraordinary parts",
  "progressive, abundant energies",
  "intense, focused nature",
  "greatly forceful energies",
    "sudden recovery ability",
  "idealistic, honorable side"};

byte * ARR szDesc[cSign+1] = {"",
  "forceful, energetic, direct, courageous",
  "practical, often skeptical and stubborn",
  "inquisitive, witty, perceptive, adaptable",
  "introspective, emotional, protective",
  "proud, gregarious, dramatic, dignified",
  "analytical, critical, modest, helpful",
  "affectionate, tolerant, often indecisive, appreciates beauty",
  "penetrating, suspicious, introspective",
  "jovial, open minded, loves freedom",
  "industrious, practical, disciplined",
  "progressive, erratic, revolutionary, idealistic, humanitarian, inventive",
  "imaginative, other worldly, impressionable"};

byte * ARR szDesire[cSign+1] = {"",
  "seeks adventure and challenge",
  "loves serenity and inner peace",
  "seeks out information",
  "wants inner and outer security",
  "desires self expression",
  "works toward perfection",
  "seeks balance, relationship, and calmness",
  "desires to transform and remove outer masks",
  "seeks meaning and new experience",
  "works for solitude and personal integrity",
  "desires individuality",
  "seeks transcendence of self"};

byte * ARR szLifeArea[cSign+1] = {"",
  "establishment of personal identity",
  "self image, self worth, and material security",
  "communicating to and receiving from the environment",
  "imagination, fantasies, inner feelings, and domestic life",
  "finding joy, pleasure, and creative expression",
  "work and feeling talented and useful",
  "personal relationships and intimacy",
  "sex, death, the occult, and other hard to grasp topics",
  "changes, exploration, and the breaking of routine",
  "career, social stature, and destiny",
  "the future, life goals, and association with friends and groups",
  "things that disrupt or cause disassociation with the personality"};

byte * ARR szInteract[cAspectInt+1] = {"",
  "is %sconnected and fused together with",
  "%sopposes and creates tension with",
  "is %sin conflict with",
  "is %sin harmony with",
  "has %sopportunity for growth, exchange, and harmony in relation with",
  "is %sdifferent from",
  "%sgets new perspectives in relation with",
  "%screates internal friction with",
  "%screates internal agitation with",
  "%screatively relates externally with",
  "%screatively relates internally with"};

byte * ARR szTherefore[cAspectInt+1] = {"",
  "Both parts are prominent in their psyche", "Balance is needed",
  "Adaptation is required by both sides", "", "",
  "They can often relate in a discordant way", "", "", "", "", ""};

/* Modifier array makes the interpretation stronger for narrower orbs. */

byte * ARR szModify[3][cAspectInt] =
  {{"always ", "always ", "irreconcilably ", "always ", "much ",
  "completely ", "often ", "often ", "often ", "often ", "often "},
  {"", "", "", "", "", "", "", "", "", "", ""},
  {"somewhat ", "somewhat ", "somewhat ", "somewhat ", "some ", "somewhat ",
  "sometimes ", "sometimes ", "sometimes ", "sometimes ", "sometimes "}};
#endif /* INTERPRET */

/* data2.c */
